### PR TITLE
Matrix product and Langevin simplification

### PR DESCRIPTION
**Pull request by fweik**
Description of changes:
 - 3x3 matrix product for Utils::Matrix
 - space to body frame conversion for linear maps
 - Some simplification of the langevin thermostat

### CODE REVIEW

**Review by jngrad — APPROVED** [review]
Can't really tell for Langevin-related modifications, but other than that, LGTM.

**Comment by fweik** [comment]
I disagree, the name for the matrix transformation was wrong ^^

**Comment by RudolfWeeber** [comment]
This creates a merge conflict with the Brownian Dynamics pr. A merge order needs to be chosen.

File: src/core/rotation.hpp [hunk 1]
```diff
@@ -57,8 +57,25 @@ inline Utils::Vector3d convert_vector_space_to_body(const Particle &p,
   return transpose(rotation_matrix(p.r.quat)) * v;
 }
 
+/**
+ * @brief Transform matrix from body- to space-fixed frame.
+ *
+ * Given a linear map represented by \f$A \in \mathbb{R}^{3 \times 3}$\f
+ * in the body-fixed frame, this returns the matrix \f$A \in \mathbb{R}^{3
+ * \times 3}$\f representing the map in the space-fixed frame. They are related
```
Comment (jngrad) [inline src/core/rotation.hpp]: `$\f` -> `\f$` twice

**Comment by fweik** [comment]
We probably should rename the vector transform functions to `convert_body_to_space`?
I can rebase the BD PR, did we decided to merge that?

**Comment by RudolfWeeber** [comment]
On Tue, Nov 12, 2019 at 02:15:40AM -0800, Florian Weik wrote:
> We probably should rename the vector transform functions to `convert_body_to_space`?
> I can rebase the BD PR, did we decided to merge that?
Yes. See notes on the offline discussion with Bogdan from the school on the PR.

**Comment by fweik** [comment]
bors r=jngrad

**Pull request merged**